### PR TITLE
Cover `<MatchCriterion>` component with tests

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose
Cover MatchCriterion component with tests

## Approach
Covered MatchCriterion component with tests

## Refs
https://issues.folio.org/browse/UIDATIMP-711

## Screenshots

image

### CODE REVIEW

**Review by IvanKryzhanovskyi — APPROVED** [review]

File: src/components/MatchCriterion/edit/MatchCriterion.test.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  existingRecordLabel: 'testExistingRecordLabel',
+  existingRecordFields: [{
+    value: 'testValue_1',
+    label: 'testLabel_1',
+  }],
+  onStaticValueTypeChange: onEachChangeMock,
+  onQualifierSectionChange: onEachChangeMock,
+  changeFormState: onEachChangeMock,
+  formValues: { profile: { matchDetails: [{ existingMatchExpression: { fields: [{ value: 'test value' }] } }] } },
+};
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/MatchCriterion.test.js:57]: Move the common fields for this and `matchCriterionWithoutQualifierType` variable to a separate variable, please

File: src/components/MatchCriterion/edit/MatchCriterion.test.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      onQualifierSectionChange={onQualifierSectionChange}
+      changeFormState={changeFormState}
+      formValues={formValues}
+    />
+  );
+
+  return renderWithIntl(renderWithFinalForm(component), translationsProperties);
+};
+
+describe('MatchCriterion edit', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/MatchCriterion.test.js:138]: Mock all sections and check that it re-renders when `existing`/`incoming` value changes and all sections are displayed

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  return renderWithIntl(renderWithFinalForm(component), translationsProperties);
+};
+
+describe('QualifierPartSection edit', () => {
+  afterAll(() => {
+    onChangeMock.mockClear();
+  });
+
+  it('should be rendered with closed additional content', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:44]:
```suggestion
  it('should be closed when section is unchecked', () => {
```

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const { container } = renderQualifierPartSection(qualifierPartSection);
+    const additionalContent = container.querySelector('.content');
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:46]:
```suggestion
    const sectionContent = container.querySelector('.content');
```

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    expect(additionalContent).toBeNull();
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:48]: Is there something like `is.not.visible`?

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  });
+
+  describe('when click on checkbox', () => {
+    it('additional content should be opened', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:52]:
```suggestion
    it('section content should be expanded', () => {
```

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  });
+
+  it('should be rendered with closed additional content', () => {
+    const { container } = renderQualifierPartSection(qualifierPartSection);
+    const additionalContent = container.querySelector('.content');
+
+    expect(additionalContent).toBeNull();
+  });
+
+  describe('when click on checkbox', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:51]:
```suggestion
  describe('when clickong on the section checkbox', () => {
```

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    it('additional content should be opened', () => {
+      const { container } = renderQualifierPartSection(qualifierPartSection);
+      const element = container.querySelector('input[type="checkbox"]');
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:54]: Rename it, please

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+      fireEvent.click(element);
+
+      const additionalContent = container.querySelector('.content');
+
+      expect(additionalContent).toBeDefined();
+    });
+  });
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:62]:
```suggestion
   describe('Qualifier part section content', () => {
     it('should have a dropdown with correct options');
  });
```

File: src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  );
+
+  return renderWithIntl(renderWithFinalForm(component), translationsProperties);
+};
+
+describe('QualifierPartSection edit', () => {
+  afterAll(() => {
+    onChangeMock.mockClear();
+  });
+
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierPartSection/QualifierPartSection.test.js:43]:
```suggestion
it('should have a correct title')
```

File: src/components/MatchCriterion/edit/QualifierSection/QualifierSection.test.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+      const element = container.querySelector('input[type="checkbox"]');
+
+      fireEvent.click(element);
+
+      const additionalContent = container.querySelector('.content');
+
+      expect(additionalContent).toBeDefined();
+    });
+  });
+});
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/edit/QualifierSection/QualifierSection.test.js:63]: Update it by analogy with the Qualifier part test comments, please

File: src/components/MatchCriterion/view/ExistingRecordSections/ExistingRecordSections.test.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+      existingRecordType={existingRecordType}
+      existingRecordFieldLabel={existingRecordFieldLabel}
+    />
+  );
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('ExistingRecordSections view', () => {
+  it('should be rendered with existing field label', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/ExistingRecordSections/ExistingRecordSections.test.js:32]:
```suggestion
  it('should render a correct value', () => {
```

File: src/components/MatchCriterion/view/ExistingRecordSections/ExistingRecordSections.test.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+      existingRecordFields={existingRecordFields}
+      existingRecordType={existingRecordType}
+      existingRecordFieldLabel={existingRecordFieldLabel}
+    />
+  );
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('ExistingRecordSections view', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/ExistingRecordSections/ExistingRecordSections.test.js:31]: Add test that it renders a correct label

File: src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('IncomingSectionStatic view', () => {
+  it('should be rendered empty', () => {
+    const { queryByText } = renderIncomingSectionStatic(emptyIncomingSectionStatic);
+
+    expect(queryByText('test text')).toBeNull();
+  });
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js:76]:
```suggestion
describe('when there is no value', () => {
    it('should be rendered empty', () => {
    const { queryByText } = renderIncomingSectionStatic(emptyIncomingSectionStatic);

    expect(queryByText('test text')).toBeNull();
  });
})
```

File: src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+  );
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('IncomingSectionStatic view', () => {
+  it('should be rendered empty', () => {
+    const { queryByText } = renderIncomingSectionStatic(emptyIncomingSectionStatic);
+
+    expect(queryByText('test text')).toBeNull();
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js:75]: Can we do smth like this?
```suggestion
    expect(valueElement).toBeNotDefined();
```

File: src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+  });
+
+  describe('when choose TEXT static value type', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js:78]:
```suggestion
  describe('when value equals to TEXT', () => {
```

File: src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+    it('should be rendered TEXT static value type', () => {
+      const { getByText } = renderIncomingSectionStatic(incomingSectionStaticWithText);
+
+      expect(getByText('test text')).toBeDefined();
+    });
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js:83]: describe('and there is a filled text field')

File: src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+
+    it('should be rendered with empty TEXT static value type', () => {
+      const { getByLabelText } = renderIncomingSectionStatic(incomingSectionStaticWithoutText);
+
+      expect(getByLabelText('No value set')).toBeDefined();
+    });
+  });
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/IncomingSectionStatic/IncomingSectionStatic.test.js:90]: describe('and the text field is not filled')

File: src/components/MatchCriterion/view/MARCFieldSection/MARCFieldSection.test.js [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+import { translationsProperties } from '../../../../../test/jest/helpers';
+
+import { MARCFieldSection } from './MARCFieldSection';
+
+const marcFieldSection = {
+  expressionDetails: {
+    fields: [{
+      label: 'field',
+      value: 'test value',
+    }],
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/MARCFieldSection/MARCFieldSection.test.js:14]: Add values for each field, please and test that it renders

File: src/components/MatchCriterion/view/MARCFieldSection/MARCFieldSection.test.js [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('MARCFieldSection view', () => {
+  it('should be rendered with label', () => {
+    const { getByText } = renderMARCFieldSection(marcFieldSection);
+
+    expect(getByText('Test label')).toBeDefined();
+  });
+});
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/MARCFieldSection/MARCFieldSection.test.js:42]: Add check that `NoValue` (simple dash `-`) component renders when there are no values

File: src/components/MatchCriterion/view/QualifierPartSection/QualifierPartSection.test.js [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+      recordFieldType={recordFieldType}
+      qualifierData={qualifierData}
+    />
+  );
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('QualifierPartSection view', () => {
+  it('should be rendered with additional content', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/QualifierPartSection/QualifierPartSection.test.js:41]:
```suggestion
describe('when there is a value')
  it('should render the section content', () => {
```

File: src/components/MatchCriterion/view/QualifierPartSection/QualifierPartSection.test.js [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+    const { container } = renderQualifierPartSection(qualifierPartSection);
+    const element = container.querySelector('.content');
+
+    expect(element).toBeDefined();
+  });
+
+  it('should be rendered without additional content', () => {
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/QualifierPartSection/QualifierPartSection.test.js:48]:
```suggestion
describe('when there is no value')
  it('should not render a section content', () => {
```

File: src/components/MatchCriterion/view/QualifierSection/QualifierSection.test.js [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+    const element = container.querySelector('.content');
+
+    expect(element).toBeDefined();
+  });
+
+  it('should be rendered without additional content', () => {
+    const { container } = renderQualifierSection(qualifierSectionWithoutQualifierType);
+    const element = container.querySelector('.content');
+
+    expect(element).toBeNull();
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/QualifierSection/QualifierSection.test.js:52]: The same here

File: src/components/MatchCriterion/view/ViewMatchCriterion.test.js [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+    expect(getByText('Incoming testIncomingRecordLabel record')).toBeDefined();
+    expect(getByText('Existing testExistingRecordLabel record')).toBeDefined();
+  });
+
+  it('should be rendered without a qualifier type', () => {
+    const { getByText } = renderViewMatchCriterion(viewMatchCriterionWithoutMatchCriterion);
+
+    expect(getByText('Existing testExistingRecordLabel record')).toBeDefined();
+    expect(getByText('Incoming testIncomingRecordLabel record')).toBeDefined();
+  });
```
Comment (mariia-aloshyna) [inline src/components/MatchCriterion/view/ViewMatchCriterion.test.js:117]: The same as for the edit component

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**